src/core/border_router/routing_manager.cpp

```diff
@@ -199,20 +222,23 @@ void RoutingManager::Stop(void)
 
     UnpublishLocalOmrPrefix();
 
-    if (mAdvertisedOnLinkPrefix != nullptr)
+    if (mIsAdvertisingLocalOnLinkPrefix)
     {
-        RemoveExternalRoute(*mAdvertisedOnLinkPrefix);
-    }
+        RemoveExternalRoute(mLocalOnLinkPrefix);
 
-    InvalidateAllDiscoveredPrefixes();
+        // Start deprecating the local on-link prefix to send
+        // a PIO with zero preferred lifetime.
+        DeprecateOnLinkPrefix();
```

Comment:
mOnLinkPrefixDeprecateTimer is stopped later in this function. How is this method effective?

Reply:
The `SendRouterAdvertisement` depends on the `mOnLinkPrefixDeprecateTimer` status to know whether it should send such PIO or not. Updated the comment to make this clearer.